test/node/codabar.test.js

```diff
@@ -0,0 +1,48 @@
+var assert = require('assert');
+var JsBarcode = require('../../bin/JsBarcode.js');
+var Canvas = require("canvas");
+
+describe('Codabar', function() {
+  it('should be able to include the encoder(s)', function () {
+    Codabar = JsBarcode.getModule("codabar");
+  });
+
+  it('should encode a string with start and stop characters', function() {
+    var enc = new Codabar("A12345B");
+    assert.equal("10110010010101011001010100101101100101010101101001011010100101010010011"
+      , enc.encode().data);
+  });
+
+  it('should add start and stop characters to a string without them', function() {
+    var enc = new Codabar("12345");
+    // should encode to "A12345A"
+    assert.equal("10110010010101011001010100101101100101010101101001011010100101011001001"
+      , enc.encode().data);
+  });
+
+  it('should return text string without start/stop characters', function() {
+    var enc = new Codabar("A12345B");
+    assert.equal("12345", enc.encode().text)
+  });
+
+  it('should warn with invalid start/stop characters', function () {
+    var enc = new Codabar("X12345Y");
+    assert.equal(false, enc.valid());
+  });
+
+  it('should warn with only a start character', function () {
+    var enc = new Codabar("X12345");
```

Comment:
The start and stop characters are marked option in order to allow a "naked" string (e.g. one without them) to be passed in. If that is the case, then the encoder assumes to add "A" as both the start and stop (see line 8)